runtime/src/bank/partitioned_epoch_rewards/calculation.rs

```diff
@@ -191,11 +183,12 @@ impl Bank {
             prev_epoch_duration_in_years,
             capitalization,
             point_value,
-        } = rewards_calculation.as_ref();
+        } = &rewards_calculation;
 
         let total_vote_rewards = vote_account_rewards.total_vote_rewards_lamports;
-        self.store_vote_accounts_partitioned(vote_account_rewards, metrics);
-        self.update_vote_rewards(vote_account_rewards);
+
+        // Cache the result for other forks
+        self.cache_epoch_rewards(&rewards_calculation);
```

Comment:
The code below this needs to be updated:

No need to read lock the stakes cache for these values since we have them above
```rust
        let (num_stake_accounts, num_vote_accounts) = {
            let stakes = self.stakes_cache.stakes();
            (
                stakes.stake_delegations().len(),
                stakes.vote_accounts().len(),
            )
        };
```

The stakes history in the stakes cache hasn't been updated yet so this will always be `0`.
```rust
        let active_stake = if let Some(stake_history_entry) =
            self.stakes_cache.stakes().history().get(prev_epoch)
        {
            stake_history_entry.effective
        } else {
            0
        };
```